extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -236,6 +239,9 @@ private Uni<TenantConfigContext> createTenantContext(Vertx vertx, OidcTenantConf
         }
 
         if (!oidcConfig.discoveryEnabled.orElse(true)) {
+            if (userInfoInjectionPointDetected && oidcConfig.userInfoPath.isPresent()) {
```

Comment:
Hi @michalvavrik I don't think this is what we agreed to. 
It should be done once, at the moment when `OidcConfigurationMetadata` has been created - which will have either discovered or configured userinfo set.
Something like:
```
if (userInfoInjectionPointDetected && metadata.getUserInfoUri() != null && oidcConfig.authentication.userInfoRequired.orElse(false)) {
    enableUserInfo(oidcConfig);
}
``` 

At the moment we set this config here but we don't know if the provider supports it and below it is good, but you never know if someone has actually explicitly disabled `user-info-required=false`

Reply:
@michalvavrik Ignore please some of my comments there, both of your updates are quite fine, but this check can be done only once in any case, the metadata is initialized with the discovered or configured userinfo address, and I think we need to avoid resetting an explicitly configured `user-info-required=false`